src/features/prompts/server/prompt-router.ts

```diff
@@ -136,6 +167,47 @@ export const promptRouter = createTRPCRouter({
         throw e;
       }
     }),
+  history: protectedProjectProcedure
+    .input(z.object({ projectId: z.string(), name: z.string() }))
+    .query(async ({ input, ctx }) => {
+      throwIfNoAccess({
+        session: ctx.session,
+        projectId: input.projectId,
+        scope: "prompts:read",
+      });
+      const prompts = await ctx.prisma.prompt.findMany({
+        where: {
+          projectId: input.projectId,
+          name: input.name,
+        },
+        orderBy: [{ version: "desc" }],
+      });
+      const users = await ctx.prisma.user.findMany({
+        select: {
+          // never select passwords as they should never be returned to the FE
+          id: true,
+          name: true,
+          email: true,
+        },
+        where: {
+          memberships: {
+            some: {
+              projectId: input.projectId,
+            },
+          },
+        },
+      });
+
+      const joinedPromptAndUsers = prompts.map((p) => {
+        const user = users.find((u) => u.id === p.createdBy);
+        if (!user) return { ...p, creator: "Unknown" };
```

Comment:
We have the "API" case as well, right? In this case I would also return API. If it is unknown, i would also log this to the console, because this should never happen.